Week-3/assignment-2.js

```diff
@@ -0,0 +1,71 @@
+console.log('--------------------');
+console.log('This is assignment 2');
+
+console.log('------------Bind------------');
+
+a=10 //Global variable 
+b=10 //Global variable 
+
+// Bind
+function multiply(a,b) {
+
+    return this.a * this.b;
+    
+}
+
+var multiplyB = multiply.bind({a:2,b:3})
+
+console.log(multiply(6,7)); //100
+console.log(multiplyB()); //6
+console.log(multiplyB(3,4)); //6
+
+/* 
+Bind method creates a new method from the method it is binded and nullifies the this keyword functionality
+ If we are not using Bind then the this keyword will take the global variable if availble and return undefined/NAN if there is no global variables. 
+ Bind restrict the values to the provided one. In function-1 multiply(6,7) the global values will be taken.
+
+ Once binded the values are resricted to the ones provided. multiplyB will alwasys give 6 as output. 
+*/
+
+// Call
+
+console.log('------------Call------------');
+
+function pesto(additionalInformation) {
+    console.log(`${this.name} have completed assignment ${this.value} ${additionalInformation}.`);
+}
+
+soumya={
```

Comment:
@soumya213 prefer using let or const, this is global variable. If type is not mentioned it goes to the global under window.

Reply:
@RajeshSivanesan Added `const` keyword along with below changes: 

Changes made: 

Added `const` keyword to: 
```
const a=10 //Global variable 
const b=10 //Global variable 
```

Added the below if-else clause to:

`function pesto(additionalInformation)` and `function pestoApplyfunction(additionalInformation)` to remove `undefined` from when `additionalInformation` value is not provided

```
if (additionalInformation) {
        console.log(`${this.name} have completed assignment ${this.value} ${additionalInformation}.`);
    }
    else{
    console.log(`${this.name} have completed assignment ${this.value}.`);
    }
```
Added `let` keyword to below objects: 

```
let soumya={
    name:'Soumya',
    value:3
}

let vijay={
    name:'Vijay',
    value:2
}
```